### PR TITLE
Made duplicate remove filter work across multiple layers

### PR DESCRIPTION
**Pull request by zerebubuth**
Extracted deduplicator to a class, makes logic easier to read. Allowed the duplicate remover to work across several layers. Requires a bit of abnormal fiddling with the layers to get it to work.

Basically, the layers are mutable when passed in. For convenience, we can return a layer and it will be re-inserted into the layers list, but we're also passed the layer list as a parameter, so can do the modifications directly. This isn't exactly idiomatic, so there's a big comment explaining what's going on, and we might want to consider follow-up changes to make this sort of thing easier to reason about.

Connects to mapzen/vector-datasource#266.

@rmarianski could you review, please?

### CODE REVIEW

File: TileStache/Goodies/VecTiles/transform.py [hunk 1]
```diff
@@ -2067,47 +2137,51 @@ def remove_duplicate_features(
...
+    deduplicator = _Deduplicator(property_keys, min_distance,
+                                 none_means_unique)
+
+    for source_layer in source_layers:
+        layer_index = -1
+        # because this post-processor can potentially modify
+        # multiple layers, and that wasn't how the return value
+        # system was designed, instead it modifies layers
+        # *in-place*. this is abnormal, and as such requires a
+        # nice big comment like this!
```
Comment (rmarianski) [inline TileStache/Goodies/VecTiles/transform.py:2149]: Should we just change the contract so that post processors can modify any number of layers? Maybe they should just return the new list?

(If so, we can do it in a separate pr)

File: TileStache/Goodies/VecTiles/transform.py [hunk 2]
```diff
@@ -2067,47 +2137,51 @@ def remove_duplicate_features(
...
+        layer['features'] = new_features
+        feature_layers[index] = layer
 
-    layer['features'] = new_features
-    return layer
+    # returning None here would normally indicate that the
+    # post-processor has done nothing. but because this
+    # modifies the layers *in-place* then all the return
+    # value is superfluous.
+    return None
```
Comment (rmarianski) [inline TileStache/Goodies/VecTiles/transform.py:2184]: Just thinking out loud, I wonder if there would be any potential efficiencies gained by consolidating all the post processors that create new label features into a single one. If so, we might want to consider that in the future.

**Comment by rmarianski** [comment]
:+1:

File: TileStache/Goodies/VecTiles/transform.py (hunk 1 above)
Comment (zerebubuth) [inline TileStache/Goodies/VecTiles/transform.py:2149]: Yup, having them return the complete new list sounds like a better option.

Comment (zerebubuth) [inline TileStache/Goodies/VecTiles/transform.py:2149]: Added #86 for this.

**Comment by rmarianski** [comment]
:+1:
